app/src/components/adminConsole/manageTeam/SnappablesTable.tsx

```diff
@@ -1,35 +1,110 @@
 import React from "react";
 import { Entity, Snap, Snappable } from "../../../types";
-import SnappableRow from "./SnappablesRow";
 import { countSnapsForUser } from "./CountSnapsForUser";
+import { useTable, useSortBy, useFlexLayout } from "react-table";
 
 const SnappablesTable = (props: {
     snappables: Entity<Snappable>[];
     currentSnaps?: Entity<Snap>[];
 }) => {
-    const rows = props.snappables
-        .sort((a: Snappable, b: Snappable) =>
-            a.fullName.localeCompare(b.fullName)
-        )
-        .map((p: Snappable) => (
-            <SnappableRow
-                key={p.id}
-                snappable={p}
-                numSnaps={countSnapsForUser(p.id, props.currentSnaps)}
-            />
-        ));
+    const defaultColumn = React.useMemo(
+        () => ({
+            width: 2, // width is used for both the flex-basis and flex-grow
+        }),
+        []
+    );
+
+    const columns = React.useMemo(
+        () => [
+            {
+                Header: "Email",
+                accessor: "email", // must match property keys on data objects
+            },
+            {
+                Header: "Full Name",
+                accessor: "fullName",
+            },
+            {
+                Header: "Username",
+                accessor: "username",
+            },
+            {
+                Header: "Snaps Received",
+                accessor: "numSnaps",
+            },
+        ],
+        []
+    );
+
+    const data = React.useMemo(
+        () =>
+            props.snappables
+                .sort((a: Snappable, b: Snappable) =>
+                    a.fullName.localeCompare(b.fullName)
+                )
+                .map((p: Snappable) => ({
+                    id: p.id,
+                    email: p.email,
+                    fullName: p.fullName,
+                    username: p.username,
+                    numSnaps: countSnapsForUser(p.id, props.currentSnaps),
+                })),
+        [props]
```

Comment:
Should we more explicitly say `props.snappables` and `props.currentSnaps` in the dependency array?
That would guard against recomputing the result should someone add an unrelated prop in future.

Reply:
Sure thing